docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+
+## Usage
+
+### Basic usage
+
+if the `webpacker.yml` file is configured as instructed [here](https://github.com/shakacode/shakapacker#configuration-and-code), With following configurations
+
+```yml
+default: &default
+    source_path: app/javascript
+    source_entry_path: packs 
+    public_root_path: public
+    public_output_path: packs
+    nested_entries: true
+# And more 
+```
+
+the directory structure will look like this
+```
+app/javascript:
+  └── packs:               # sets up webpack entries
+  │   └── application.js   # references ../src/my_component.js
+  │   └── application.css
+  └── src:                 # any directory name is fine. Referenced files need to be under source_path
+  │   └── component.js
+  └── stylesheets:
+  │   └── my_styles.css
+  └── images:
+      └── logo.svg
+public/packs                # webpack output
+```
+
+Now, To automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with `react_component` and `react_component_hash` helpers, Create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
+  │   │ └── ...
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   └── A_ComponentOne.jsx
+  │   └── B
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   │ └── B_ComponentOne.jsx
+  │   │   │ └── B_ComponentTwo.jsx
+public/packs                
+```
+
+To register react component, It is no longer needed to create a pack entry and manually register it by calling `ReactOnRails.register`. With automatically generated packs, you can directly use `A_ComponentOne`, `B_ComponentOne` and `B_ComponentTwo` in rails view using:
+
+```erb
+    <%= react_component("A_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentTwo", {}, auto_load_bundle: true) %>    
+```
+
+If `A_ComponentOne` uses multiple HTML strings for server rendering, `react_component_hash` view helper can be used on rails view as illustrated below. Read more about [`react_component_hash` here](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component_hash).
+
+```erb
+<% A_ComponentOne = react_component_hash("A_ComponentOne", 
+                                             prerender: true,
+                                             auto_load_bundle: true
+                                             props: {}
+                                          ) %>   
+<% content_for :title do %>
+   <%= A_ComponentOne['title'] %>
+<% end %>
+<%= A_ComponentOne["componentHtml"] %>
+```
+
+`auto_load_bundle` option can be omitted, if `config/initializers/react_on_rails` is configured with `config.auto_load_bundle = true`.
```

Comment:
Not just omitted, you may also make it false explicitly, right? I would say instead that 

> the default value of the `auto_load_bundle` parameter can be specified by setting `config.auto_load_bundle` in `config/initializers/react_on_rails.rb`.